biz.ganttproject.core/src/test/kotlin/biz/ganttproject/core/chart/scene/gantt/TableSceneBuilderTest.kt

```diff
@@ -0,0 +1,40 @@
+package biz.ganttproject.core.chart.scene.gantt
+
+import org.junit.Test
+import org.mockito.Mockito.*
+
+import biz.ganttproject.core.chart.canvas.Canvas
+import biz.ganttproject.core.chart.scene.gantt.TableSceneBuilder.*
+
+class TableSceneBuilderTest {
+  @Test
+  fun `test width when max width comes from large indent row`() {
+    val rowHeight = 10
+    val horizontalOffset = 5
+    val rows = listOf(
+      Row(10, "", 0), Row(1, "", 10), Row(4, "", 5),
+    )
+    test(rowHeight, horizontalOffset, rows)
+  }
+
+  @Test
+  fun `test width when max width comes from large text width row`() {
+    val rowHeight = 10
+    val horizontalOffset = 5
+    val rows = listOf(
+      Row(12, "", 0), Row(1, "", 10), Row(4, "", 5),
+    )
+    test(rowHeight, horizontalOffset, rows)
+  }
+
+  private fun test(rowHeight: Int, horizontalOffset: Int, rows: List<Row>) {
+    val canvas = spy(Canvas())
```

Comment:
spying is okay, but perhaps it is easier to just pass the real simple Canvas object and check its contents after scene building. Because we know the primitives which are expected to be there.

Reply:
Canvas' primitives lists are private. Didn't want to add getters because immutable lists in java are awkward

Reply:
I see